### ISSUE TITLE
Feature request: Option to move Knowledge skills into the other skills colum and have the 3rd purely for the picture

### ISSUE THREAD

**Issue by Zitchas** [chummer5a/chummer5a#1986]
This is a feature request for the character sheet creator

### Desired behaviour
That Knowledge Skills appear as a group at the end of the Active skills (and the two columns re-balanced accordingly), and the third column is thus just the character picture.

This would solve the aesthetic unpleasantness of having a large chunk of white space at the bottom of the two active skills lists because the third column has both all the knowledge skills and the character picture.

Ideally, probably as a "this is a tall picture" checkbox of some kind next to the picture (alongside "this is the mugshot" option) that controls whether or not the picture has the 3rd column for itself or shares it with the knowledge skills

**Comment by DelnarErsike** [comment]
So, I'm going to give our hand away and reveal that we don't really have any active Chummer5 devs who are good with what is essentially web design that is used to construct character sheets. Most of the latest sheet updates have been done by me, and I'm just a monkey with a keyboard and internet access when it comes to HTML and CSS.

Unless you can find someone to actively work on that part of Chummer5, or you do it yourself, this isn't going to show up anytime soon.

**Comment by Zitchas** [comment]
OK. I've done a little html, xml, and css stuff, but not much. I'd be happy to see if I can figure something out. Is there any documentation on how the rest of it interacts with the html? I can't promise anything, but I'd love to be able to contribute something beyond reporting issues.

**Comment by DelnarErsike** [comment]
No documentation, unfortunately. Sheets are stored in the /sheets folder as .xsl scripts and .xslt template scripts. These use XSL to programmatically take values outputted from clsCharacter::Print() to generate HTML and CSS information that ends up as the different character sheets.

Chummer5 uses a WebBrowser control to preview the sheets (WebBrowser uses whatever version of Internet Explorer the user has installed), and the "Save as PDF" option in the current nightly uses a WebKit-based rendering engine.

**Comment by Zitchas** [comment]
It looks like the active skills get split somewhere else and fed to the .xslt as skills1, skills2, and knowledge skills are fed into skills3. Can you confirm that, or am I just missing something?

**Comment by Zitchas** [comment]
OK, thanks!

 I found the sheets and the secondary pages that put together skills1,2, and 3. Slowly piecing together how this all works. 

(For the record, Commlinks, Formatted Text-Only, and Text-Only  reports an error attempting to load it, "the variable or parameter is 'lang.PhysicalTrack1' is either not defined or out of scope." I'm not sure if I broke something or if that's actually an issue, so I'm not reporting them separately until I've re-installed to check, though.)

**Comment by chummer5a** [comment]
Looks like incorrect strings were used there, properly speaking they should be using $lang.PhysicalTrack instead. It's something on our end, I'll push something to fix it in a bit.

**Comment by DelnarErsike** [comment]
I've made some changes to the Shadowrun 5 character sheets that should show up in 5.195.52, though you can get them to show up if you clone the GitHub repository and compile Chummer5 yourself. One of those changes is to how the main mugshot shows up in the sheet.

The change isn't specifically what you asked for, but its functionality is similar enough that you may want to have a look at it and close this issue if those changes make this request irrelevant.

**Comment by Lochabar** [comment]
When I reworked the character sheets to make it easier to create foreign language versions without having to maintain multiple versions of the main code, I kept the essentially the same format of the original sheets.

To create a new sheet combining the skills I would do something like this:
1) Copy "Shadowrun 5 (Rating greater 0).xsl" as, say, "Shadowrun 5 (Combined Skills).xsl"

2) In "Shadowrun 5 (Combined Skills).xsl". The code has

<xsl:variable name="MinimumRating" select="1"/>
modify this to

<xsl:variable name="CombineSkills" select="Y"/>

3) In "Shadowrun 5 set.xslt" there is a section of code setting local versions of global variables. Add the new variable:
 
  <xsl:variable name="CombineSkills" select="N"/>

4) Copy "xs.SkillsGroupedAssorted.xslt" as, say, "xs.SkillsGroupedCombined.xslt"

5) "xs.SkillsGroupedCombined.xslt" will contain the main code.
The select statement in templates "skills1" and "skills2" will probably become
select="skills/skill[rating &gt; 0 or total &gt; 0]"/> (i.e. removing knowledge = 'False')
In template "skills3", the knowledge skills select statements would be removed, leaving just the "call-template" to mugshots remaining.
